sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ServiceBusMessageJavaDocCodeSamples.java

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.core.amqp.models.AmqpAnnotatedMessage;
+import com.azure.core.amqp.models.AmqpMessageBodyType;
+
+/**
+ * java doc sample for  {@link ServiceBusMessage}.
+ */
+public class ServiceBusMessageJavaDocCodeSamples {
+    /**
+     * Copy {@link AmqpAnnotatedMessage}.
+     */
+    public void copyServiceBusReceivedMessage() {
+        ServiceBusReceivedMessage sourceServiceBusReceivedMessage = null;
+        // BEGIN: com.azure.messaging.servicebus.ServiceBusMessage.copyServiceBusMessage
+        AmqpMessageBodyType bodyType = sourceServiceBusReceivedMessage.getAmqpAnnotatedMessage().getBody().getBodyType();
+        ServiceBusMessage copyMessage = null;
+        switch (bodyType) {
+            case DATA:
+                copyMessage = new ServiceBusMessage(sourceServiceBusReceivedMessage);
+                break;
+            case SEQUENCE:
+            case VALUE:
+                throw new RuntimeException("Body type not supported yet.");
+            default:
+                throw new RuntimeException("Body type not valid.");
+        }
```

Comment:
To copy a received message, why does the user have to check the body type? This should be taken care of in the constructor of `ServiceBusMessage`.

Reply:
Actually, we will not create a SBRM with any other type except DATA. So User should not face this problem in copy constructor.